pre-commit-hooks/fix-local-includes.sh

```diff
@@ -16,6 +16,16 @@ else
     main_src_dirs=$(find ./src -maxdepth 1 -type d -exec basename {} \; | paste -sd '|' | sed 's/|/\\|/g')
 fi
 
+# Get a list of all header filenames in src/ and tests/ (recursively, unique)
+all_local_headers=$(find ./src ./tests -type f \( -name '*.hpp' -o -name '*.h' \) -exec basename {} \; | sort -u)
+
+# Generate a sed script for batch replacements
+sed_script=$(mktemp)
+for header in $all_local_headers; do
+    escaped_header=$(printf '%s\n' "$header" | sed 's/[\/&]/\\&/g')
+    echo "s|#include[[:space:]]*[\"<]$escaped_header[\">]|#include \"$header\"|g" >> "$sed_script"
+done
```

Comment:
Is it possible to make replacements directly, instead of generating code to make replacements?